src/Client.cpp

```diff
@@ -153,7 +152,8 @@ void Client::readFromClient() {
 		size_t parsedSize = _header.parseBuffer(buffer);
 		if (parsedSize < static_cast<size_t>(bytes_received)) {
 			// save rest to bodybuffer
-			_bodyBuffer = std::string(&buffer[parsedSize]);
+			for (; parsedSize < static_cast<size_t>(bytes_received); parsedSize++)
+				_bodyBuffer.push_back(buffer[parsedSize]);
```

Comment:
`_bodyBuffer = std::string(&buffer[parsedSize], parsed_size - bytes_received);`
this safes new allocations